spotbugs/src/main/java/edu/umd/cs/findbugs/ba/type/TypeAnalysis.java

```diff
@@ -888,8 +888,6 @@ private ExceptionSet computeThrownExceptionTypes(BasicBlock basicBlock) throws D
                     Type throwType = frame.getTopValue();
                     if (throwType instanceof ObjectType) {
                         exceptionTypeSet.addExplicit((ObjectType) throwType);
-                    } else if (throwType instanceof ExceptionObjectType) {
-                        exceptionTypeSet.addAll(((ExceptionObjectType) throwType).getExceptionSet());
```

Comment:
One more idea is that, switch the order of if-blocks like below. It could be similar to the intent of the author the commit 3b31f36597c84e4dfe9cb0bd1e631c309e6bc660.

```java
                    if (throwType instanceof ExceptionObjectType) {
                        exceptionTypeSet.addAll(((ExceptionObjectType) throwType).getExceptionSet());
                    } else if (throwType instanceof ObjectType) {
                        exceptionTypeSet.addExplicit((ObjectType) throwType);
                    } else {
```

However, the behavior of `addAll()` is not the same as `addExplicit()`, so I put a higher priority on keeping the current behavior.